politician_analysis/src/scraper.py

```diff
@@ -0,0 +1,126 @@
+                    yield value
+                else:
+                    stack.append(value)
+        elif isinstance(current_item, list):
+            for value in current_item:
+                stack.append(value)
+
+
+def download_comments(youtube_id, sort_by=SORT_BY_RECENT, sleep=.1):
+    
+    """Download comments from youtube URL, send ajax request for dynamic webpages.
+    Load the response JSON file and sorted the comments if it's not sorted by popularity.
+    Lastly, save comment information as a dictionary.
+    
+
+    Parameters:
+    youtube_id : youtube id for the selected video.
+    sort_by : Has SORT_BY_RECENT and SORT_BY_POPULAR based on customers settings.
+    sleep : wait for a given number of seconds.
+
+    Returns:
+    comment data: A dictionary with selected comments information.
+    """
+    
+    
+    session = requests.Session()
+    session.headers['User-Agent'] = USER_AGENT
+
+    response = session.get(YOUTUBE_URL.format(youtube_id=youtube_id))
+
+    if 'uxe=' in response.request.url:
+        session.cookies.set('CONSENT', 'YES+cb', domain='.youtube.com')
+        response = session.get(YOUTUBE_URL.format(youtube_id=youtube_id))
+
+    html = response.text
+    version = find_value(html, 'client.version', 4, '\'')
+    api_key = find_value(html, 'INNERTUBE_API_KEY', 3)
+
+    data = json.loads(find_value(html, 'var ytInitialData = ', 0, '};') + '}')
+    section = next(search_dict(data, 'itemSectionRenderer'), None)
+    renderer = next(search_dict(section, 'continuationItemRenderer'), None) if section else None
+    if not renderer:
+        return
+
+    needs_sorting = sort_by != SORT_BY_POPULAR
+    continuations = [renderer['continuationEndpoint']]
+    while continuations:
+        continuation = continuations.pop()
+        response = ajax_request(session, continuation, version, api_key)
+
+        if not response:
+            break
+        if list(search_dict(response, 'externalErrorMessage')):
+            raise RuntimeError('Error returned from server: ' + next(search_dict(response, 'externalErrorMessage')))
+
+        if needs_sorting:
+            sort_menu = next(search_dict(response, 'sortFilterSubMenuRenderer'), {}).get('subMenuItems', [])
+            if sort_by < len(sort_menu):
+                continuations = [sort_menu[sort_by]['serviceEndpoint']]
+                needs_sorting = False
+                continue
+            raise RuntimeError('Failed to set sorting')
+
+        actions = list(search_dict(response, 'reloadContinuationItemsCommand')) + \
+                  list(search_dict(response, 'appendContinuationItemsAction'))
+        for action in actions:
+            for item in action.get('continuationItems', []):
+                if action['targetId'] == 'comments-section':
+                    # Process continuations for comments and replies.
+                    continuations[:0] = [ep for ep in search_dict(item, 'continuationEndpoint')]
```

Comment:
To be more readable, we can change it to continuations = [ep for ep in search_dict(item, 'continuationEndpoint')] + continuations